routers/web/repo/pull.go

```diff
@@ -732,19 +756,35 @@ func ViewPullFiles(ctx *context.Context) {
 	if fileOnly && (len(files) == 2 || len(files) == 1) {
 		maxLines, maxFiles = -1, -1
 	}
-	diffOptions := &gitdiff.DiffOptions{
-		BeforeCommitID:     startCommitID,
-		AfterCommitID:      endCommitID,
-		SkipTo:             ctx.FormString("skip-to"),
-		MaxLines:           maxLines,
-		MaxLineCharacters:  setting.Git.MaxGitDiffLineCharacters,
-		MaxFiles:           maxFiles,
-		WhitespaceBehavior: gitdiff.GetWhitespaceFlag(ctx.Data["WhitespaceBehavior"].(string)),
+
+	var diffOptions *gitdiff.DiffOptions
+
+	// show only a single commit for this pr
+	if willShowSingleCommit {
+		diffOptions = &gitdiff.DiffOptions{
+			AfterCommitID:      endCommitID,
+			SkipTo:             ctx.FormString("skip-to"),
+			MaxLines:           maxLines,
+			MaxLineCharacters:  setting.Git.MaxGitDiffLineCharacters,
+			MaxFiles:           maxFiles,
+			WhitespaceBehavior: gitdiff.GetWhitespaceFlag(ctx.Data["WhitespaceBehavior"].(string)),
+		}
+	} else {
+		// show full PR diff
+		diffOptions = &gitdiff.DiffOptions{
```

Comment:
There are many duplicated code. Maybe 

```suggestion
		diffOptions = &gitdiff.DiffOptions{
			AfterCommitID:      endCommitID,
			SkipTo:             ctx.FormString("skip-to"),
			MaxLines:           maxLines,
			MaxLineCharacters:  setting.Git.MaxGitDiffLineCharacters,
			MaxFiles:           maxFiles,
			WhitespaceBehavior: gitdiff.GetWhitespaceFlag(ctx.Data["WhitespaceBehavior"].(string)),
		}
		if xxxx {
		  diffOptions.BeforeCommitID = xxx
		  }
```
is better?